src/app/detail/ui/TaskEditor.tsx

```diff
@@ -152,23 +154,28 @@ export const TaskEditor = ({
 
       <Box mt="12px" sx={{ height: '100%' }}>
         <Tapwrite
-          uploadFn={async (file, tiptapEditorUtils) => {
-            const newBlob = await upload(file.name, file, {
-              access: 'public',
-              handleUploadUrl: '/api/upload',
-            })
-            tiptapEditorUtils.setImage(newBlob.url as string)
-          }}
           content={updateDetail}
           getContent={handleDetailChange}
           readonly={userType === UserType.CLIENT_USER}
           editorClass="tapwrite-details-page"
           placeholder="Add description..."
+          uploadFn={async (file, tiptapEditorUtils) => {
+            const supabaseActions = new SupabaseActions()
+            const fileName = generateRandomString(file.name)
+            const signedUrl: ISignedUrlUpload = await getSignedUrlUpload(fileName)
+            const filePayload = await supabaseActions.uploadAttachment(file, signedUrl, task_id)
```

Comment:
the name `uploadAttachment` can lead to confusion as we have a separate section for attachments. This is better called something like `inlineFileUpload`.